.github/workflows/release.yml

```
@@ -0,0 +1,58 @@
+name: Build and Release
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        goos: [linux]
+        goarch: [amd64, arm64]
+
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '1.17'
+
+    - name: Build
+      run: |
+        GOOS=${{ matrix.goos }} GOARCH=${{ matrix.goarch }} go build -o myapp ./agent/cmd/runner
+
+    - name: Upload artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: myapp-${{ matrix.goos }}-${{ matrix.goarch }}
+        path: myapp
+
+  release:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download artifact
+      uses: actions/download-artifact@v2
+      with:
+        name: myapp-linux-amd64
+
+    - name: Set up SLSA Generator
+      uses: slsa-framework/slsa-github-generator@v1
+      with:
+        artifact-path: myapp-linux-amd64
+
+    - name: Create GitHub Release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: v1.0.0
+        release_name: Release v1.0.0
+        draft: false
```

from SecurityExpert
The use of `GITHUB_TOKEN` is generally safe for most operations, but it's important to ensure that it has the least privileges necessary for the task. In this workflow, it's used to create a release, which is a common practice. However, ensure that no other sensitive operations are performed with this token without proper permissions.

```suggestion
env:
  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```